draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -885,6 +885,129 @@ Type     Description                   Used In
+    that some payloads that are not relevant to multiple key exchanges may be omitted for brevity.
+    </t>
+
+    <section title="No Additional Key Exchange Used" anchor="sample-exchanges-none-selected">
+      <t>The initiator proposes two sets of optional additional key exchanges, but the responder does not
+      support any of them.  The responder chooses NONE for each set and consequently, IKE_INTERMEDIATE
+      exchange does not takes place and the exchange proceeds to IKE_AUTH phase.  The resulting keying
+      materials are the same as those derived with <xref target="RFC7296"/>.</t>
+
+      <figure><artwork align="center"><![CDATA[
+Initiator                     Responder
+------------------------------------------------------------------------
+HDR(IKE_SA_INIT), SAi1(.. AKE*...), --->
+KEi1, Ni, N(IKEV2_FRAG_SUPPORTED),
+N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+  Proposal #1
+    Transform ECR (ID = ENCR_AES_GCM_16,
+                   256-bit key)
+    Transform PRF (ID = PRF_HMAC_SHA2_512)
+    Transform KE (ID = Curve25519)
+    Transform AKE1 (ID = PQ_KEM_1)
+    Transform AKE1 (ID = PQ_KEM_2)
+    Transform AKE1 (ID = NONE)
+    Transform AKE2 (ID = PQ_KEM_3)
+    Transform AKE2 (ID = PQ_KEM_4)
+    Transform AKE2 (ID = NONE)
+                         <--- HDR(IKE_SA_INIT), SAr1(.. AKE*...),
+                              KEr1, Nr, N(IKEV2_FRAG_SUPPORTED),
+                              N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+                                Proposal #1
+                                  Transform ECR (ID = ENCR_AES_GCM_16,
+                                                 256-bit key)
+                                  Transform PRF (ID = PRF_HMAC_SHA2_512)
+                                  Transform KE (ID = Curve25519)
+                                  Transform AKE1 (ID = NONE)
+                                  Transform AKE2 (ID = NONE)
+
+HDR(IKE_AUTH), SK{ IDi, AUTH, SAi2, TSi, TSr } --->
+                         <--- HDR(IKE_AUTH), SK{ IDr, AUTH, SAr2,
+                              TSi, TSr }
+      ]]></artwork></figure>
+    </section>
+
+    <section title="Additional Key Exchange for CHILD SA only" anchor="sample-exchanges-ake-child-sas">
+      <t>The exchanges below show that the initiator does not propose the use of additional key exchanges
+      to establish an IKE SA, but they are required in order to establish a CHILD SA.  Note that the
+      initiator does not advertise N(INTERMEDIATE_EXCHANGE_SUPPORTED) in the IKE_SA_INIT request.</t>
+
+      <figure><artwork align="center"><![CDATA[
+Initiator                     Responder
+------------------------------------------------------------------------
+HDR(IKE_SA_INIT), SAi1(.. AKE*...), --->
+KEi1, Ni, N(IKEV2_FRAG_SUPPORTED)
+                         <--- HDR(IKE_SA_INIT), SAr1(.. AKE*...),
+                              KEr1, Nr, N(IKEV2_FRAG_SUPPORTED)
+
```

Comment:
Should we also remove AKE* from SA payload in this case?

Reply:
Ah good catch, thanks. Yes, it should be removed.